### PR TITLE
Phase11

### PR DESCRIPTION
**Pull request by nazari02105**

### CODE REVIEW

File: Phase8/EFCore/WebApi/Controllers/SearchController.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+        [HttpPost]
+        public IActionResult PostDocuments([FromBody] Dictionary<string, string> fileContents)
+        {
+            _invertedIndex.AddDocuments(fileContents);
+            return Ok(new
+            {
+                fileContents
+            });
+        }
```
Comment (soroushtaslimi) [inline Phase8/EFCore/WebApi/Controllers/SearchController.cs:31]: users are probably not supposed to send the documents to server over http request, but they only ask for search results from server (imagine sending a ton of documents over internet!).

File: Phase8/EFCore/SearchEngine/InvertedIndex.cs [hunk 2]
```diff
@@ -8,29 +8,47 @@ namespace SearchEngine
     public class InvertedIndex : IInvertedIndex
     {
         private readonly IDatabaseMap<string,string> _context;
+        private readonly ITokenizer _tokenizer;
         
-        public InvertedIndex(IDatabaseMap<string,string> context)
+        public InvertedIndex(IDatabaseMap<string,string> context,ITokenizer tokenizer)
```
Comment (soroushtaslimi) [inline Phase8/EFCore/SearchEngine/InvertedIndex.cs]: It's not important but there should be an space between comma and ITokenizer :)

**Review by soroushtaslimi — CHANGES_REQUESTED** [review]
Please review the comments

File: Phase8/EFCore/WebApi/Controllers/SearchController.cs (hunk 1 above)
Comment (mohammad258852) [reply Phase8/EFCore/WebApi/Controllers/SearchController.cs:31]: you're right. Actual documents should be added with ConsoleApp to the database but users can also add some documents to the available documents

**Review by matindaghyani — APPROVED** [review]

File: Phase8/EFCore/SearchEngine/Document.cs [hunk 3]
```diff
@@ -5,8 +5,7 @@ namespace SearchEngine
 {
     public class Document
     {
-        [Key]
-        public string Name { get; init; }
+        [Key] public string Name { get; init; }
```
Comment (alrz1999) [inline Phase8/EFCore/SearchEngine/Document.cs:8]: ```[Key]```
اینو IDE پیشنهاد داد که بیاد تو یه خط؟

**Review by alrz1999 — APPROVED** [review]

File: Phase8/EFCore/SearchEngine/Document.cs (hunk 3 above)
Comment (mohammad258852) [reply Phase8/EFCore/SearchEngine/Document.cs:8]: بله موقع درست کردن ایندنت‌ها و فاصله‌ها پیشنهاد داد که تو یه خط باشه

**Review by soroushtaslimi — APPROVED** [review]

**Pull request merged**